PagingWithNetworkSample/app/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/repository/inMemory/byItem/InMemoryByItemRepository.kt

```diff
@@ -16,54 +16,31 @@
 
 package com.android.example.paging.pagingwithnetwork.reddit.repository.inMemory.byItem
 
-import androidx.annotation.MainThread
-import androidx.lifecycle.switchMap
-import androidx.paging.Config
-import androidx.paging.toLiveData
+import androidx.paging.PagingConfig
+import androidx.paging.PagingDataFlow
 import com.android.example.paging.pagingwithnetwork.reddit.api.RedditApi
-import com.android.example.paging.pagingwithnetwork.reddit.repository.Listing
 import com.android.example.paging.pagingwithnetwork.reddit.repository.RedditPostRepository
-import com.android.example.paging.pagingwithnetwork.reddit.vo.RedditPost
-import java.util.concurrent.Executor
+import kotlinx.coroutines.CoroutineDispatcher
 
 /**
- * Repository implementation that returns a Listing that loads data directly from the network
- * and uses the Item's name as the key to discover prev/next pages.
+ * Repository implementation that that loads data directly from the network and uses the Item's name
+ * as the key to discover prev/next pages.
  */
 class InMemoryByItemRepository(
         private val redditApi: RedditApi,
-        private val networkExecutor: Executor) : RedditPostRepository {
-    @MainThread
-    override fun postsOfSubreddit(subReddit: String, pageSize: Int): Listing<RedditPost> {
-        val sourceFactory = SubRedditDataSourceFactory(redditApi, subReddit, networkExecutor)
-
-        // We use toLiveData Kotlin ext. function here, you could also use LivePagedListBuilder
-        val livePagedList = sourceFactory.toLiveData(
-                // we use Config Kotlin ext. function here, could also use PagedList.Config.Builder
-                config = Config(
-                        pageSize = pageSize,
-                        enablePlaceholders = false,
-                        initialLoadSizeHint = pageSize * 2),
-                // provide custom executor for network requests, otherwise it will default to
-                // Arch Components' IO pool which is also used for disk access
-                fetchExecutor = networkExecutor)
-
-        val refreshState = sourceFactory.sourceLiveData.switchMap {
-            it.initialLoad
-        }
-        return Listing(
-                pagedList = livePagedList,
-                networkState = sourceFactory.sourceLiveData.switchMap {
-                  it.networkState
-                },
-                retry = {
-                    sourceFactory.sourceLiveData.value?.retryAllFailed()
-                },
-                refresh = {
-                    sourceFactory.sourceLiveData.value?.invalidate()
-                },
-                refreshState = refreshState
+        private val networkDispatcher: CoroutineDispatcher
+) : RedditPostRepository {
+    override fun postsOfSubreddit(subReddit: String, pageSize: Int) = PagingDataFlow(
+            PagingConfig(
+                    pageSize = pageSize,
+                    enablePlaceholders = false,
+                    initialLoadSize = pageSize * 2
```

Comment:
this override seems unnecessary (not sure why it's there in the first place), suggest just removing it - we already expand initial load size based on pageSize

Reply:
Done